From 5e13be4770b77c6450bec718bfa7ab683d053824 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Mon, 19 Dec 2016 12:54:44 +0800
Subject: [PATCH 1/1] irqchip: irq-xilinx-intc: fix a non-proper function definition by convention

Since local irq should be mapped irq number, so we change irq into mirq
to avoid the confliction with irq parameter as irq_flow_handler_t.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/irqchip/irq-xilinx-intc.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/irqchip/irq-xilinx-intc.c b/drivers/irqchip/irq-xilinx-intc.c
index f895b91..9df53d5 100644
--- a/drivers/irqchip/irq-xilinx-intc.c
+++ b/drivers/irqchip/irq-xilinx-intc.c
@@ -157,12 +157,12 @@ static const struct irq_domain_ops xintc_irq_domain_ops = {
 	.map = xintc_map,
 };
 
-static void intc_handler(struct irq_desc *desc)
+static void intc_handler(int irq, struct irq_desc *desc)
 {
 	struct irq_chip *chip = irq_desc_get_chip(desc);
 	struct intc *local_intc =
 		irq_data_get_irq_handler_data(&desc->irq_data);
-	int irq;
+	int mirq;
 
 	pr_debug("intc_handler: input irq = %d\n", desc->irq_data.irq);
 	chained_irq_enter(chip, desc);
@@ -171,10 +171,10 @@ static void intc_handler(struct irq_desc *desc)
 	 * ignore the irq input as we now need to query the AXI interrupt
 	 * controller to see which interrupt is active
 	 */
-	irq = get_irq(local_intc);
-	while (irq != -1U) {
-		generic_handle_irq(irq);
-		irq = get_irq(local_intc);
+	mirq = get_irq(local_intc);
+	while (mirq != -1U) {
+		generic_handle_irq(mirq);
+		mirq = get_irq(local_intc);
 	}
 
 	chained_irq_exit(chip, desc);
-- 
2.9.3

